src/EditorFeatures/Core/Shared/Preview/PreviewWorkspace.cs

```diff
@@ -27,8 +27,7 @@ public PreviewWorkspace(HostServices hostServices)
         public PreviewWorkspace(Solution solution)
             : base(solution.Workspace.Services.HostServices, WorkspaceKind.Preview)
         {
-            var oldSolution = this.CurrentSolution;
-            var newSolution = this.SetCurrentSolution(solution);
+            var (oldSolution, newSolution) = this.SetCurrentSolution(solution);
```

Comment:
basic idea is that SetCurrentSOlution now returns *both* the newSolution it set the workspace to, and the immediately prior 'oldSolution' it updated from.  These are then the suitable solutions that should be issued in the workspace event stream.